rwengine/src/script/modules/GTA3ModuleImpl.inl

```diff
@@ -6610,70 +6610,64 @@ void opcode_0249(const ScriptArguments& args, const ScriptModel model) {
 }
 
 /**
-    @brief %3d% = create_phone_at %1d% %2d%
+    @brief %3d% = create_payphone_at %1d% %2d%
 
     opcode 024a
     @arg coord Coordinates
-    @arg phone Handle
+    @arg payphone Handle
 */
-void opcode_024a(const ScriptArguments& args, ScriptVec2 coord, ScriptPhone& phone) {
-    RW_UNIMPLEMENTED_OPCODE(0x024a);
-    RW_UNUSED(coord);
-    RW_UNUSED(phone);
-    RW_UNUSED(args);
+void opcode_024a(const ScriptArguments& args, const ScriptVec2 coord, ScriptPayphone& payphone) {
+    payphone = args.getWorld()->createPayphone(coord);
 }
 
 /**
-    @brief text_phone_repeatedly %1d% %2g%
+    @brief text_payphone_repeatedly %1d% %2g%
 
     opcode 024b
-    @arg phone 
+    @arg payphone 
     @arg arg2 
 */
-void opcode_024b(const ScriptArguments& args, const ScriptPhone phone, const ScriptString arg2) {
+void opcode_024b(const ScriptArguments& args, const ScriptPayphone payphone, const ScriptString arg2) {
     RW_UNIMPLEMENTED_OPCODE(0x024b);
-    RW_UNUSED(phone);
+    RW_UNUSED(payphone);
     RW_UNUSED(arg2);
     RW_UNUSED(args);
 }
 
 /**
-    @brief text_phone %1d% %2g%
+    @brief text_payphone %1d% %2g%
 
     opcode 024c
-    @arg phone 
-    @arg arg2 
+    @arg payphone 
+    @arg text 
 */
-void opcode_024c(const ScriptArguments& args, const ScriptPhone phone, const ScriptString arg2) {
-    RW_UNIMPLEMENTED_OPCODE(0x024c);
-    RW_UNUSED(phone);
-    RW_UNUSED(arg2);
+void opcode_024c(const ScriptArguments& args, const ScriptPayphone payphone, const ScriptString text) {
     RW_UNUSED(args);
+    payphone->setMessageAndStartRinging(text);
 }
 
 /**
-    @brief   phone_text_been_displayed %1d%
+    @brief   payphone_text_been_displayed %1d%
 
     opcode 024d
-    @arg phone 
+    @arg payphone 
 */
-bool opcode_024d(const ScriptArguments& args, const ScriptPhone phone) {
+bool opcode_024d(const ScriptArguments& args, const ScriptPayphone payphone) {
     RW_UNIMPLEMENTED_OPCODE(0x024d);
-    RW_UNUSED(phone);
+    RW_UNUSED(payphone);
     RW_UNUSED(args);
     return false;
 }
 
 /**
-    @brief disable_phone %1d%
+    @brief disable_payphone %1d%
 
     opcode 024e
-    @arg phone 
+    @arg payphone 
 */
-void opcode_024e(const ScriptArguments& args, const ScriptPhone phone) {
-    RW_UNIMPLEMENTED_OPCODE(0x024e);
-    RW_UNUSED(phone);
+void opcode_024e(const ScriptArguments& args, const ScriptPayphone payphone) {
     RW_UNUSED(args);
+    payphone->disable();
```

Comment:
Does it work (I mean payphone is const)?

Reply:
Pointer to the object is unchangeable, not the object itself, object method declaration implies that object can be mutated (no const).